File: Selenium/Activities/Activity14.java
package Activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.WebDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;


public class Activity14 {
    public static void main(String[] args) throws InterruptedException
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver newDriver = new FirefoxDriver();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        newDriver.get("https://training-support.net/selenium/tables");
        System.out.println("Title is: "+newDriver.getTitle());
        List<WebElement> rowsEle = newDriver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        List<WebElement> colsEle = newDriver.findElements(By.xpath("//table[@id='sortableTable']/thead/tr/th"));
        System.out.println("No. of rows are " + rowsEle.size() + "and columns are "+colsEle.size());
        System.out.println("Value at 2nd row, 2nd column:" + newDriver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]")).getText());
        newDriver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th")).click();
        System.out.println("Value at 2nd row, 2nd column after sorting:" + newDriver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]")).getText());
        List<WebElement> footEle = newDriver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        System.out.println("Footer content:");
        for (WebElement cell : footEle)
            System.out.print(cell.getText() + " ");
        Thread.sleep(300);
        newDriver.quit();

    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://www.training-support.net/selenium/tables")
    print("Page title is: ", driver.title)
    columns = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/thead/tr/th")
    print("Number of columns: ", len(columns))
    rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    print("Number of rows: ", len(rows))

    print("Cell value at second row and second column: ", driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]").text)

    driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead/tr/th[1]").click()
    print("Cell value at second row and second column: ", driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]").text)
    driver.quit()